var/spack/repos/builtin/packages/opencv/package.py

```diff
@@ -2,399 +2,1105 @@
+        "opencl_svm",
+        "openclamdblas",
+        "openclamdfft",
+        "openni",
+        "openni2",
+        "pvapi",
+        "ueye",
+        "va",
+        "va_intel",
+        "ximea",
+        "xine",
     ]
 
+    # components and modules with the same name
+    # used in `def cmake_args(self)`
+    component_and_module = ["freetype", "julia", "matlab"]
+
     for component in components:
-        variant(component, default=False,
-                description='Include {0} support'.format(component))
+        variant(
+            component,
+            default=False,
+            description="Include {0} support".format(component),
+        )
 
     # Other (variants)
-    variant('shared', default=True,
-            description='Enables the build of shared libraries')
-    variant('powerpc', default=False, description='Enable PowerPC for GCC')
-    variant('fast-math', default=False,
-            description='Enable -ffast-math (not recommended for GCC 4.6.x)')
-    variant('nonfree', default=False, description='Enable non-free algorithms')
-
-    variant('contrib', default=False, description='Adds in code from opencv_contrib.')
-    contrib_vers = [
-        '3.1.0', '3.2.0', '3.3.0', '3.3.1', '3.4.0', '3.4.1', '3.4.3', '3.4.4',
-        '3.4.5', '3.4.6', '3.4.12', '4.0.0', '4.0.1', '4.1.0', '4.1.1',
-        '4.1.2', '4.2.0', '4.5.0', '4.5.1', '4.5.2'
-    ]
-    for cv in contrib_vers:
-        resource(name="contrib",
-                 git='https://github.com/opencv/opencv_contrib.git',
-                 tag="{0}".format(cv),
-                 when='@{0}+contrib'.format(cv))
+    variant("shared", default=True, description="Enables the build of shared libraries")
+    variant("powerpc", default=False, description="Enable PowerPC for GCC")
+    variant(
+        "fast-math",
+        default=False,
+        description="Enable -ffast-math (not recommended for GCC 4.6.x)",
+    )
+    variant("nonfree", default=False, description="Enable non-free algorithms")
 
     # Required (dependencies)
-    depends_on('cmake@3.5.1:', type='build')
-    depends_on('python@2.7:2.8,3.2:', type='build')
-    depends_on('zlib@1.2.3:')
-
-    # OpenCV modules (dependencies)
-    depends_on('java', when='+java_bindings_generator')
-    depends_on('ant', when='+java_bindings_generator', type='build')
-    extends('python', when='+python2')
-    depends_on('python@2.7:2.8', when='+python2', type=('build', 'link', 'run'))
-    depends_on('py-setuptools', when='+python2', type='build')
-    depends_on('py-numpy', when='+python2', type=('build', 'run'))
-    extends('python', when='+python3')
-    depends_on('python@3.2:', when='+python3', type=('build', 'link', 'run'))
-    depends_on('py-setuptools', when='+python3', type='build')
-    depends_on('py-numpy', when='+python3', type=('build', 'run'))
-    depends_on('ffmpeg', when='+videoio')
-    depends_on('mpi', when='+videoio')
+    depends_on("gmake", type="build")
```

Comment:
Is GNU Make actually required, or can it be any Make? Unless there's a GNUmakefile somewhere, I would remove this dep.

Reply:
I guess probably not. The output said that it found `gmake`, which I interpreted to mean that it was looking specifically for `gmake`. On my system, `gmake` is a symlink to `make` and I did not test with `bmake`. I removed the dependency.